azbfs/url_file.go

```diff
@@ -198,3 +204,50 @@ func (f FileURL) FlushData(ctx context.Context, fileSize int64, contentMd5 []byt
 		md5InBase64, nil, nil, nil, nil, nil, nil, nil,
 		nil, nil, &overrideHttpVerb, nil, nil, nil, nil)
 }
+
+// Renames the file to the provided destination
+func (f FileURL) Rename(ctx context.Context, destinationFileSystem *string, destinationPath *string) (FileURL, error) {
+
+	// If the destinationFileSystem is not provided, use the current filesystem
+	fileSystemName := destinationFileSystem
+	if fileSystemName == nil || *fileSystemName == "" {
+		fileSystemName = &f.fileSystemName
+	}
+
+	renameSource := "/" + f.fileSystemName + "/" + f.path
+
+	urlParts := NewBfsURLParts(f.fileClient.URL())
+	urlParts.FileSystemName = *fileSystemName
+	urlParts.DirectoryOrFilePath = *destinationPath
+
+	destinationFileURL := NewFileURL(urlParts.URL(), f.fileClient.Pipeline())
+
+	_, err := destinationFileURL.fileClient.Create(ctx, *fileSystemName, *destinationPath, PathResourceNone, nil, PathRenameModeLegacy,
+		nil, nil, nil, nil, nil, nil, nil, nil, nil, &renameSource, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+		nil, nil, nil, nil, nil, nil, nil)
+
+	if err != nil {
+		return FileURL{}, err
+	}
+
+	return destinationFileURL, nil
+}
+
+func (f FileURL) SetAccessControlList(ctx context.Context, accessControlList *string, group *string, owner *string) (*PathUpdateResponse, error) {
```

Comment:
Can you check if mode and additional principals are set in acl through portal, calling a setACL removes those or not. In blobfuse I am hitting this so better to handle it here from begining.

Reply:
Admittedly, this is pulled straight from my gen 2 ACLs PR-- But would you mind adding options bags?

Reply:
I'll just wait for your acls PR to merge and I can just remove this.

Reply:
@vibhansa-msft I believe this will set the ACL (not update) just as it does in blobfuse so we will have to prior getACL and or it